packages/code-studio/src/main/AppMainContainer.test.jsx

```diff
@@ -91,95 +129,84 @@ it('listens for widgets properly', () => {
     callback = cb;
   });
 
-  const wrapper = makeAppMainContainer({ session });
+  render(getAppMainContainer({ session }));
 
-  expect(wrapper.state('widgets')).toEqual(expect.arrayContaining([]));
   expect(session.connection.subscribeToFieldUpdates).toHaveBeenCalled();
 
+  const panelsButton = screen.getByRole('button', { name: 'Panels' });
+  userEvent.click(panelsButton);
+
+  expect(screen.getByText('No bound variables found.')).toBeTruthy();
+
   callback({
     created: [TABLE_A],
     removed: [],
     updated: [],
   });
 
-  expect(wrapper.state('widgets')).toEqual(expect.arrayContaining([TABLE_A]));
+  expect(screen.getByRole('button', { name: 'a' })).toBeTruthy();
 
   callback({
     created: [TABLE_B],
     removed: [],
     updated: [TABLE_A],
   });
 
-  expect(wrapper.state('widgets')).toEqual(
-    expect.arrayContaining([TABLE_A, TABLE_B])
-  );
+  expect(screen.getByRole('button', { name: 'a' })).toBeTruthy();
+  expect(screen.getByRole('button', { name: 'b' })).toBeTruthy();
 
   callback({
     created: [],
     removed: [TABLE_A],
     updated: [],
   });
 
-  expect(wrapper.state('widgets')).toEqual(expect.arrayContaining([TABLE_B]));
+  expect(screen.queryByRole('button', { name: 'a' })).toBeNull();
+  expect(screen.getByRole('button', { name: 'b' })).toBeTruthy();
 
   callback({
     created: [TABLE_A],
     removed: [TABLE_B],
     updated: [],
   });
 
-  expect(wrapper.state('widgets')).toEqual(expect.arrayContaining([TABLE_A]));
-
-  wrapper.unmount();
+  expect(screen.queryByRole('button', { name: 'b' })).toBeNull();
+  expect(screen.getByRole('button', { name: 'a' })).toBeTruthy();
 });
 
 describe('hydrates widgets correctly', () => {
   const localDashboardId = DEFAULT_DASHBOARD_ID;
   let session = null;
-  let wrapper = null;
-
-  beforeEach(() => {
-    session = makeSession();
-    wrapper = makeAppMainContainer({ session });
-  });
-
   it('hydrates empty props with defaults', () => {
-    const result = wrapper.instance().hydrateDefault({}, localDashboardId);
-    expect(result).toEqual(
-      expect.objectContaining({
-        localDashboardId: DEFAULT_DASHBOARD_ID,
-      })
-    );
-    expect(result).not.toEqual(expect.objectContaining({ fetch: expect.any }));
+    session = makeSession();
```

Comment:
You could still keep the `beforeEach` block with the `makeSession` in it, as it makes it clear the session is always initialized to the same thing.